MobiusCore/Source/EffectHandlers/EffectCallback.swift

```diff
@@ -54,17 +64,33 @@ public final class EffectCallback<Output> {
         }
     }
 
+    /// Send a number of events to the Mobius loop, then `end()` this callback.
+    /// Note: After calling this function, all operations on this object will be no-ops.
+    /// - Parameter outputs: The events which should be sent to the loop.
     public func end(with outputs: Output...) {
         end(with: outputs)
     }
 
+    /// Send a number of events to the Mobius loop and `end()` this callback.
+    /// Note: After calling this function, all operations on this object will be no-ops.
+    /// - Parameter outputs: The events which should be sent to the loop.
     public func end(with outputs: [Output]) {
+        var runOnEnd = false
+        _ended.mutate {
+            runOnEnd = !$0
+            $0 = true
+        }
         for output in outputs {
```

Comment:
I think this needs to be inside the `if runOnEnd` condition? Would be good with a test that does two `end(with:)` calls, it looks to me like that would today lead to both sets being sent.

Reply:
Good catch